Framework/Core/include/Framework/AnalysisHelpers.h

```diff
@@ -103,32 +153,28 @@ struct WritingCursor<soa::Table<ORIGIN, PC...>> {
 };
 
 /// Helper to define output for a Table
-template <typename T>
+template <o2::soa::is_table T>
 struct OutputForTable {
   using table_t = T;
-  using metadata = typename aod::MetadataTrait<table_t>::metadata;
+  using metadata = aod::MetadataTrait<o2::aod::Hash<T::ref.desc_hash>>::metadata;
 
   static OutputSpec const spec()
   {
-    return OutputSpec{OutputLabel{metadata::tableLabel()}, metadata::origin(), metadata::description(), metadata::version()};
+    return OutputSpec{OutputLabel{o2::aod::Hash<T::ref.label_hash>::str}, o2::aod::Hash<T::ref.origin_hash>::origin, o2::aod::description(o2::aod::Hash<T::ref.desc_hash>::str), T::ref.version};
   }
 
   static OutputRef ref()
   {
-    return OutputRef{metadata::tableLabel(), metadata::version()};
+    return OutputRef{o2::aod::Hash<T::ref.label_hash>::str, T::ref.version};
   }
 };
 
 /// This helper class allows you to declare things which will be created by a
 /// given analysis task. Notice how the actual cursor is implemented by the
 /// means of the WritingCursor helper class, from which produces actually
 /// derives.
-template <typename T>
-requires(!std::is_same_v<void, typename aod::MetadataTrait<T>::metadata>) struct Produces : WritingCursor<typename soa::PackToTable<aod::MetadataTrait<T>::metadata::origin(), typename T::table_t::persistent_columns_t>::table> {
-};
-
-template <template <o2::framework::OriginEnc, typename...> class T, o2::framework::OriginEnc ORIGIN, typename... C>
-struct Produces<T<ORIGIN, C...>> : WritingCursor<typename soa::PackToTable<ORIGIN, typename T<ORIGIN, C...>::table_t::persistent_columns_t>::table> {
+template <producable T>
```

Comment:
`Producible`  (with capital P), maybe?

Reply:
I used std's approach to naming concepts, so `is_producible` would be the better choice.